packages/unmock-core/src/runner/index.ts

```diff
@@ -63,8 +62,16 @@ export default (
     runnerConfiguration.minItems = Math.floor(Math.random() * 2 ** (i % 5)); // 2^5 seems enough for min items/length
     try {
       const r = await (fn ? fn(intermediaryDoneCallback) : undefined);
+      // reset histories
+      Object.entries(backend.serviceStore.services).forEach(([_, service]) => {
+        service.spy.resetHistory();
+      });
       res.push(r);
     } catch (e) {
+      // reset histories
+      Object.entries(backend.serviceStore.services).forEach(([_, service]) => {
+        service.spy.resetHistory();
```

Comment:
Would be maybe better to just add a method `serviceStore.resetSpies()` 🙂

Reply:
Also, could use a `finally` block?

Reply:
I added a finally block. We can add `serviceStore.resetSpies()` in another commit.
In general, @idantene was right that the code has big technical debt on the topmost layer because of the whole `backend` zombie alive from the `jsdom` days. If a generous soul wants to refactor during Hacktober fest so that the code is more readable and maintanable given its intent, that would be awesome!

Reply:
Yes, I also think it should have been renamed to `Interceptor` in those days when it was only a MITM wrapper to ensure it only does interception and not so much more.